.github/workflows/require_label.yml

```
@@ -22,3 +22,13 @@ jobs:
             feature
             documentation
             no changelog
+  do-not-merge:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'do not merge') }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: prevent merge
+        run: |
+          tee "${GITHUB_STEP_SUMMARY}" <<'EOF'
+          This PR is labeled as `do not merge`, remove the label to make this check pass.
+          EOF
+          exit 1
```

```suggestion
          echo "::error This PR is labeled as `do not merge`, remove the label to make this check pass."
          exit 1
```

usually github won't link to the summary of the job on failure, maybe add an error message to the log with error prefix.

---

Hm, I wanted to have the message both in the job logs as well as in the summary, but I think `::error` won't play nice there. Let me rewrite it.